### PR TITLE
blueprints: add Flycast video/blueprint

### PR DESCRIPTION
**Pull request by Xe**
### Summary of changes

Copies https://youtu.be/PR0rzkwKwFA into the docs

### Related Fly.io community and GitHub links

https://community.fly.io/t/run-private-applications-with-flycast/20387/1

### Notes

Like #1575, but using youtube-nocookie.com for the embed so that people don't get cookies set on their machines.

### CODE REVIEW

File: blueprints/private-applications-flycast.html.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+However, with a fully private app, all connections go directly to the Machines via their .internal addresses, so you have to keep them running 24/7 to maintain connectivity. This is fine for services like database engines where you want them to be running all the time, but what about an admin panel? You want your admin panel to be separate from your main app so that users can ever get into it, even by accident, but you also want it to shut down when it’s not in use.
+
+Flycast exists for this middle category of apps. With Flycast, your apps are only visible over your organization’s private network, but any traffic to them goes through the proxy so they can turn on when you need them and turn off when you don’t. This allows your administrative panels to be physically separate so that users can’t access them.
+
+When you want to connect to an app via Flycast, you connect to appname.flycast.
```
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:34]:
```suggestion
When you want to connect to an app via Flycast, you connect to `appname.flycast`.
```

File: blueprints/private-applications-flycast.html.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+Ollama doesn’t ship with authentication by default. When you create an instance of Ollama, anyone can access it without entering in a username, password, or API key. This is fine for running your models on your own computer; but it means that if you expose it to the internet, anyone can use it and run models whenever they want. This could rack up your bill infinitely.
+
+This is where Flycast comes in. Flycast lets you run a copy of Ollama on your private network so that you and your apps can access it, but nobody else. Flycast also lets you have the platform turn off your Ollama server when you’re not using it, which will save you money. This fits into that middle ground case that Flycast covers perfectly.
+
+## Prerequisites
+
+In order to get started, you need to have the following:
+
+- A [fly.io](http://fly.io) account
+- Flyctl installed ([https://fly.io/docs/flyctl/install/](https://fly.io/docs/flyctl/install/))
```
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:57]:
```suggestion
- `flyctl` installed ([https://fly.io/docs/flyctl/install/](https://fly.io/docs/flyctl/install/))
```

File: blueprints/private-applications-flycast.html.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+The links are in the description.
+
+If you want to interact with your Flycast apps from your computer, like an Ollama instance, you’ll need to jack into your private network with WireGuard. The link for how to do that is in the description.
+
+## Steps
+
+Create a new folder on your computer called ollama. This is where we’ll put the Ollama configuration. Open a terminal in that folder and run the fly launch command:
```
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:65]:
```suggestion
Create a new folder on your computer called `ollama`. This is where we’ll put the Ollama configuration. Open a terminal in that folder and run the fly launch command:
```

File: blueprints/private-applications-flycast.html.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+```
+fly launch --from https://github.com/fly-apps/ollama-demo --no-deploy
+```
+
+This command creates a new fly app from the ollama-demo template and tells the flyctl command to not deploy it after you create the app. If we don’t do this, then the platform will create public IPv4 and IPv6 addresses, which will make this a public app. The name you choose when you create your app will be used to connect to your app over Flycast.
```
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:71]:
```suggestion
This command creates a new fly app from the [ollama-demo template](https://github.com/fly-apps/ollama-demo) and tells the flyctl command to not deploy it after you create the app. If we don’t do this, then the platform will create public IPv4 and IPv6 addresses, which will make this a public app. The name you choose when you create your app will be used to connect to your app over Flycast.
```

File: blueprints/private-applications-flycast.html.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+This app only has one IP address: a private Flycast IPv6 address. If had public IP addresses, it'd look like this:
+
+```
+$ fly ips list -a recipeficator
+VERSION	IP                    	TYPE              	REGION	CREATED AT
+v6     	2a09:8280:1::37:7312:0	public (dedicated)	global	May 30 2024 13:51
+v4     	66.241.124.113        	public (shared)   	      	Jan 1 0001 00:00
+```
+
+Now that we've proven it's private, let’s open an interactive shell machine to play around with Flycast. Create the shell machine with `fly machine run`:
```
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:104]:
```suggestion
Now that we've proven it's private, let’s open an interactive shell Machine to play around with Flycast. Create the shell Machine with `fly machine run`:
```

File: blueprints/private-applications-flycast.html.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+root@e784127b51e083:/#
+```
+
+The Ubuntu image we chose is very minimal, so we need to install a few tools such as ping, curl, and dig:
+
+```
+# apt update && apt install -y curl iputils-ping dnsutils
+```
+
+My app is named xe-ollama, so let’s look up its `.flycast` address with `nslookup xe-ollama.flycast`:
```
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:117]:
```suggestion
My app is named`xe-ollama`, so let’s look up its `.flycast` address with `nslookup xe-ollama.flycast`:
```

File: blueprints/private-applications-flycast.html.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+real	0m0.043s
+user	0m0.003s
+sys		0m0.003s
+```
+
+And if you wait a few moments, it’ll spin back down.
+
+### Running Llama 3
+
+Now that we’ve set up Ollama and demonstrated the platform turning it off and on for you, let’s run Llama 3. Exit out of that shell machine with control-D so we can make a new one with the Ollama client installed.
```
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:171]:
```suggestion
Now that we’ve set up Ollama and demonstrated the platform turning it off and on for you, let’s run Llama 3. Exit out of that shell Machine with control-D so we can make a new one with the Ollama client installed.
```

File: blueprints/private-applications-flycast.html.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+categories:
+  - networking
+date: 2024-06-17
+---
+
+<center><iframe width="600" height="315" src="https://www.youtube-nocookie.com/embed/PR0rzkwKwFA" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></center><br />
+
+## Intro
+
+A lot of the time your applications are made to be public and shared with the world. Sometimes you need to be more careful. When you deploy your apps on Fly.io, you get a private network for your organization. This lets your applications in other continents contact eachother like they were in the same room.
```
Comment (catflydotio) [inline blueprints/private-applications-flycast.html.md:16]:
```suggestion
A lot of the time your applications are made to be public and shared with the world. Sometimes you need to be more careful. When you deploy your apps on Fly.io, you get a private network for your organization. This lets your applications in other continents contact each other like they were in the same room.
```

File: blueprints/private-applications-flycast.html.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+# ollama run llama3 "Why is the sky blue?"
+```
+
+It took a moment for Ollama to get ready and download the image, then it downloaded it and answered your question. Once it’s been idle for a moment, the platform will turn Ollama back off.
+
+## Conclusion
+
+And there we go! We've covered what Flycast is, why you'd want to use it, and set up an instance of Ollama to show it off.
+
+You can use Flycast with any application that listens over HTTP or TCP without any modification to your code. UDP is a bit more tricky due to the fact that there's no sessions, but there's documentation you can follow to do this. The link is in the description.
```
Comment (catflydotio) [inline blueprints/private-applications-flycast.html.md:197]: UDP and Flycast actually _don't_ work together, right?

**Review by catflydotio — APPROVED** [review]
Sorry, didn't see I was tagged here.

Suggested correcting a typo that Vale caught, and I think the UDP statement needs a correction. Aside from that, 🚢

File: blueprints/private-applications-flycast.html.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+## Steps
+
+Create a new folder on your computer called `ollama`. This is where we’ll put the Ollama configuration. Open a terminal in that folder and run the fly launch command:
+
+```
+fly launch --from https://github.com/fly-apps/ollama-demo --no-deploy
+```
+
+This command creates a new fly app from the [ollama-demo template](https://github.com/fly-apps/ollama-demo) and tells the flyctl command to not deploy it after you create the app. If we don’t do this, then the platform will create public IPv4 and IPv6 addresses, which will make this a public app. The name you choose when you create your app will be used to connect to your app over Flycast.
```
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:71]:
```suggestion
This command creates a new fly app from the [`ollama-demo` template](https://github.com/fly-apps/ollama-demo) and tells the flyctl command to not deploy it after you create the app. If we don’t do this, then the platform will create public IPv4 and IPv6 addresses, which will make this a public app. The name you choose when you create your app will be used to connect to your app over Flycast.
```

File: blueprints/private-applications-flycast.html.md (hunk 9 above)
Comment (Xe) [inline blueprints/private-applications-flycast.html.md:197]:
```suggestion
```

**Comment by andie787** [comment]
@xe don't worry in this case about Vale check failing. It does this sometimes with older branches. Merge whenever!

**Pull request merged**